app/src/main/java/fr/free/nrw/commons/upload/SpinnerLanguagesAdapter.kt

```diff
@@ -0,0 +1,105 @@
+package fr.free.nrw.commons.upload
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import androidx.annotation.LayoutRes
+import androidx.core.os.ConfigurationCompat
+import fr.free.nrw.commons.R
+import fr.free.nrw.commons.utils.BiMap
+import fr.free.nrw.commons.utils.LangCodeUtils
+import kotlinx.android.extensions.LayoutContainer
+import kotlinx.android.synthetic.main.row_item_languages_spinner.*
+import org.apache.commons.lang3.StringUtils
+import java.util.*
+
+class SpinnerLanguagesAdapter constructor(
+    context: Context,
+    private val selectedLanguages: BiMap<*, String>
+) : ArrayAdapter<Any?>(context, -1) {
+
+    private val languageNamesList: List<String>
+    private val languageCodesList: List<String>
+
+    init {
+        val sortedLanguages =
+            Locale.getAvailableLocales().map(::Language).sortedBy { it.locale.displayName }
```

Comment:
Does our codestyle mandate a specific line length? I find that this sort of formatting tends to hinder readability instead of helping it, but if the codestyle itself specifies this then I could be wrong.

Reply:
Codestyle specifies it, [link](https://github.com/google/styleguide/blob/305ac8725a166ed42e3f5dd3f80d6de2cf840ef1/intellij-java-google-style.xml#L28)